### PR TITLE
update funktions with two Maps

### PR DESCRIPTION
**Pull request by rnaiser**

### CODE REVIEW

**Comment by rnaiser** [comment]
Bug war bei: einzelbeleg --> Tab/Reiter Textkrikit und wenn man dann neu anlegt und Bemerkung ausfüllt kommt ein error:
der sql string war nicht richtig z.b 
INSERT INTO einzelbeleg_textkritik SET EinzelbelegID = '1', EditionID = 'NULL', HandschriftID = 'NULL', Variante = '', Bemerkung = 'Peter Pan'   Broblem war das NULL in "" war nun ist das nicht mehr der fall: z.b INSERT INTO einzelbeleg_textkritik SET EinzelbelegID = '1', EditionID = '195', HandschriftID = NULL, Variante = 'ew4tr', Bemerkung = 'Halloween'

File: src/main/java/de/uni_tuebingen/ub/nppm/db/AbstractBase.java [hunk 1]
```diff
@@ -260,10 +261,46 @@ protected static String buildAndConditions(Map<String, String> andConditions) {
...
+
+            if (andCondition.getKey().equals("QuelleBisJahrhundert") || andCondition.getKey().equals("QuelleVonJahrhundert") || andCondition.getKey().equals("VonJahrhundert") || andCondition.getKey().equals("BisJahrhundert") ) {
+                query.setParameter(andCondition.getKey(), andCondition.getValue(), StringType.INSTANCE);
+            } else {
+                query.setParameter(andCondition.getKey(), andCondition.getValue());
+            }
         }
     }
 
+     public static void updateMap(String table, Map<String, String> attributesAndValues, Map<String, String> andConditions) throws Exception {
```
Comment (mtrojan-ub) [inline src/main/java/de/uni_tuebingen/ub/nppm/db/AbstractBase.java]: You can rename this function to "update" as well & simply use function overloading this way.

File: src/main/java/de/uni_tuebingen/ub/nppm/db/AbstractBase.java [hunk 2]
```diff
@@ -260,10 +261,46 @@ protected static String buildAndConditions(Map<String, String> andConditions) {
     //Hilsfunktion zum setzen der Parameter
     protected static void registerAndConditions(Map<String, String> andConditions, NativeQuery query) {
         for (Map.Entry<String, String> andCondition : andConditions.entrySet()) {
-            query.setParameter(andCondition.getKey(), andCondition.getValue());
+
+            if (andCondition.getKey().equals("QuelleBisJahrhundert") || andCondition.getKey().equals("QuelleVonJahrhundert") || andCondition.getKey().equals("VonJahrhundert") || andCondition.getKey().equals("BisJahrhundert") ) {
```
Comment (mtrojan-ub) [inline src/main/java/de/uni_tuebingen/ub/nppm/db/AbstractBase.java]: Why is this necessary? a generic function like this should not have special hardcoded handling for field names. If it is necessary to handle certain columns in a special way, this should be passed to the function as parameter as well.

File: src/main/java/de/uni_tuebingen/ub/nppm/db/AbstractBase.java [hunk 3]
```diff
@@ -260,10 +261,46 @@ protected static String buildAndConditions(Map<String, String> andConditions) {
...
+                i++;
+
+            }
+
+            sql += buildAndConditions(andConditions);
+
+            NativeQuery query = session.createNativeQuery(sql);
+
+
+            registerAndConditions(attributesAndValues, query);
```
Comment (mtrojan-ub) [inline src/main/java/de/uni_tuebingen/ub/nppm/db/AbstractBase.java]: attributesAndValues cannot be treated the same way as andConditions, so if you want to do it like this you will need 2 separate functions.

File: src/main/java/de/uni_tuebingen/ub/nppm/db/AbstractBase.java (hunk 1 above)
Comment (rnaiser) [reply src/main/java/de/uni_tuebingen/ub/nppm/db/AbstractBase.java]: done: look at commit: improvement of function update

File: src/main/java/de/uni_tuebingen/ub/nppm/db/AbstractBase.java (hunk 2 above)
Comment (rnaiser) [reply src/main/java/de/uni_tuebingen/ub/nppm/db/AbstractBase.java]: done: look at commit : improvement of function update

File: src/main/java/de/uni_tuebingen/ub/nppm/db/AbstractBase.java (hunk 3 above)
Comment (rnaiser) [reply src/main/java/de/uni_tuebingen/ub/nppm/db/AbstractBase.java]: done: look at commit: improvement of function update
